src/main/java/org/opensearch/neuralsearch/query/AgenticSearchQueryBuilder.java

```diff
@@ -224,4 +241,26 @@ public String getWriteableName() {
     public String fieldName() {
         return NAME;
     }
+
+    private String extractQueryContent(String agentResponse) throws IOException {
+        XContentParser parser = XContentType.JSON.xContent().createParser(null, LoggingDeprecationHandler.INSTANCE, agentResponse);
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+
+        XContentParser.Token token;
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME && "query".equals(parser.currentName())) {
+                parser.nextToken(); // Move to the query value
+
+                // Use XContentBuilder to extract the inner query as JSON string
+                XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
+                builder.copyCurrentStructure(parser);
+                return builder.toString();
+            } else {
+                parser.skipChildren();
```

Comment:
wonder why skipChildren here? will this cause pitfall?

Reply:
Thrown an exception if the first field is not "query"